src/components/Widgets/TableWidget.tsx

```diff
@@ -0,0 +1,40 @@
+import { Box } from '@mui/material';
+import { DataGridPro } from '@mui/x-data-grid-pro';
+import { useRef } from 'react';
+import { GenericWidget } from './GenericWidget';
+import { TableWidgetProps } from './widgetConfig';
+
+export const TableWidget = ({
+  columns,
+  dataGridProps,
+  ...props
+}: TableWidgetProps) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const maxHeight = containerRef.current?.clientHeight ?? 300;
+
+  return (
+    <GenericWidget {...props} ref={containerRef}>
+      <Box
+        sx={({ spacing }) => ({
+          height: '100%',
```

Comment:
May not matter but this would be more consistent with what you did in other places.
```css
  height: 1
```

Reply:
The 1 here would be a reference to MUI's theme.spacing which is not what we are going for.